RecoHI/HiEvtPlaneAlgos/src/EvtPlaneProducer.cc

```diff
@@ -209,15 +221,144 @@ class EvtPlaneProducer : public edm::stream::EDProducer<> {
   double maxpt_;
   double minvtx_;
   double maxvtx_;
+  int flatnvtxbins_;
+  double flatminvtx_;
+  double flatdelvtx_;
+  double dzdzerror_;
+  double d0d0error_;
+  double pterror_;
+  double chi2perlayer_;
   double dzerr_;
+  double dzdzerror_pix_;
   double chi2_;
+  int nhitsValid_;
   int FlatOrder_;
   int NumFlatBins_;
   double nCentBins_;
   double caloCentRef_;
   double caloCentRefWidth_;
   int CentBinCompression_;
+  int cutEra_;
   HiEvtPlaneFlatten *flat[NumEPNames];
+  TrackStructure track;
+
+  edm::ESWatcher<HeavyIonRcd> hiWatcher;
+  edm::ESWatcher<HeavyIonRPRcd> hirpWatcher;
+
+  void fillHF(TrackStructure track, double vz, int bin) {
+    double minet = minet_;
+    double maxet = maxet_;
+    for (int i = 0; i < NumEPNames; i++) {
+      if (EPDet[i] != HF)
+        continue;
+      if (minet_ < 0)
+        minet = minTransverse[i];
+      if (maxet_ < 0)
+        maxet = maxTransverse[i];
+      if (track.et < minet)
+        continue;
+      if (track.et > maxet)
+        continue;
+      if (EPEtaMin2[i] == EPEtaMax2[i]) {
+        if (track.eta < EPEtaMin1[i])
+          continue;
+        if (track.eta > EPEtaMax1[i])
+          continue;
+      } else {
+        if (track.eta < EPEtaMin1[i])
+          continue;
+        if (track.eta > EPEtaMax2[i])
+          continue;
+        if (track.eta > EPEtaMax1[i] && track.eta < EPEtaMin2[i])
+          continue;
+      }
+      double w = track.et;
+      if (loadDB_)
+        w = track.et * flat[i]->etScale(vz, bin);
+      if (EPOrder[i] == 1) {
+        if (MomConsWeight[i][0] == 'y' && loadDB_) {
+          w = flat[i]->getW(track.et, vz, bin);
+        }
+      }
+      rp[i]->addParticle(w, track.et, sin(EPOrder[i] * track.phi), cos(EPOrder[i] * track.phi), track.eta);
+    }
+  };
+
+  void FillCastor(TrackStructure track, double vz, int bin) {
```

Comment:
from Aug 18 https://github.com/cms-sw/cmssw/pull/31129#discussion_r472304344
was meant to apply to all methods in the PR, especially the newly added

start local variables and data members with lower case letter. Capitalization is reserved for class names.
```suggestion
  void fillCastor(const TrackStructure& track, double vz, int bin) {
```